lib/vector-core/src/event/metric/tags.rs

```diff
@@ -377,18 +377,38 @@ impl ByteSizeOf for TagValueSet {
 
 impl<'de> Deserialize<'de> for TagValueSet {
     fn deserialize<D: Deserializer<'de>>(de: D) -> Result<Self, D::Error> {
-        // Deserialize from a single string only
-        let s = String::deserialize(de)?;
-        Ok(Self::from([s]))
+        #[derive(Deserialize)]
+        #[serde(untagged)]
+        enum Variants {
```

Comment:
Just for curiosity and I'm guessing the answer is going to be "no" - are there any performance advantages to declaring this enum outside the function body ?

Reply:
Most likely no. The only possible differences I can imagine is if the change in the localization affected either inlining or code layout in a predictable way.